_layouts/news.html

```diff
@@ -3,20 +3,57 @@
 # SPDX-License-Identifier: CC-BY-SA-4.0
 layout: default
 ---
+
+{%- assign this_year = page.year | append: "" -%}
+
 {%- assign newest_url = site.pages | where: "layout", "news" | map: "name" | sort | pop | last -%}
 {%- assign newest_year = newest_url | split: "." | first -%}
-<div class="row">
- <div class="col">
-  <h1>
-    News about
-    <a href={{ "/" | relative_url }}>seL4</a>
-    and the
-    <a href={{ "/Foundation" | relative_url }}>seL4 Foundation</a>
-  </h1>
- </div>
- <div class="col">
-{% assign url = "/news/" | relative_url -%}
-Older News:  {% for i in (2020..newest_year) -%}<a href="{{ url }}{{ i }}">{{ i }}</a> {% endfor %}
- </div>
+<div class="row" style="margin-left: 0.3em;">
+  <div class="col">
+    <h1>
+      News about
+      <a href="{{ '/' | relative_url }}">seL4</a>
+      and the
+      <a href="{{ '/Foundation' | relative_url }}">seL4 Foundation</a>
+      {%- unless newest_year == this_year %} &mdash; {{ page.year }} {%- endunless %}
+    </h1>
+  </div>
+  <div class="col">
+{%- assign url = "/news/" | relative_url %}
+    News from {%- if newest_year == this_year %} older {%- else %} other {%- endif %} years:
+{%- for i in (2020..newest_year) -%}
+{%-   unless i == page.year %}
+      <a href="{{ url }}{{ i }}">{{ i }}</a>
+{%- else %}
+     {{ i }}
+{%- endunless -%}
+{%- endfor %}
+  </div>
 </div>
-{{ content }}
+
+{%  for news_item in site.news-items reversed -%}
+{%-   assign date = news_item.date | date: "%d %b %Y" -%}
```

Comment:
According to [this](https://simpleit.rocks/ruby/jekyll/collections/jekyll-collections-versus-posts/) site, collections use file creation time for their `date` property unless overruled in front matter, while posts use the filename. Have you checked if the dates are correct?

You may want to use posts instead of a collection for this reason.

Reply:
I have checked, the dates are correct. You also do get an error message if the date is not correctly formatted, e.g. if you set the template file name to `9999-99-99-template.md` generation fails. It's possible that if there is no date in the name and no date in the front matter it falls back to file creation time, but we're not doing that.

Posts by default get rendered as single files, collections are exactly for our use case, so I think this one is right.